rust/src/partitions.rs

```diff
@@ -85,12 +85,16 @@ impl<'a> TryFrom<&'a str> for DeltaTablePartition<'a> {
     type Error = DeltaTableError;
 
     fn try_from(partition: &'a str) -> Result<Self, DeltaTableError> {
-        match partition.find('=') {
-            Some(position) => Ok(DeltaTablePartition {
+        let separator = "=";
+        match (
+            partition.find(separator),
+            partition.matches(separator).count(),
```

Comment:
nitpick, the [split](https://doc.rust-lang.org/std/primitive.str.html#method.split) method will probably work better for this use-case so we won't need to search the string twice for the same separator.

Reply:
Oh! But do we need to check if there is only one `=` in `partition`? Or just ignoring it by using `split` method (and get the wrong value)?

Reply:
depends on your use-case, if we only want to split on the first `=`, then we can use [splitn](https://doc.rust-lang.org/std/primitive.str.html#method.splitn).

Reply:
Oh got it!